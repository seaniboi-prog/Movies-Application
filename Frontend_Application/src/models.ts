export interface Movie {
    id: number;
    key: string;
    name: string;
    description: string;
    genres: string[];
    rate: string;
    length: string;
    img: string;
}