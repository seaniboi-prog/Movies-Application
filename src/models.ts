export interface Movie {
    id: number;
    key: string;
    name: string;
    desc: string;
    genres: string[];
    rate: string;
    length: string;
    img_src: string;
}