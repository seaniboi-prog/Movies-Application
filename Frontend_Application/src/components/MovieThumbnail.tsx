import React from 'react'
import { useNavigate } from 'react-router-dom'
import { Movie } from '../models';
import './mainPage.scss';

interface Props {
    movie: Movie;
}

const MovieThumbnail:React.FC<Props> = ({movie}:Props) => {

    const navigate = useNavigate();

    const navigateToDetail = () => {
        navigate('/details', {state:{id: movie.id}})
    }

    return (
    <div className='movie__thumbnail' data-testid="movieThumbnail">
            <div className='movie__box' data-testid="movieBox" onClick={() => navigateToDetail()}>
                <img
                    src={process.env.PUBLIC_URL + `movie_images/${movie.img}`}
                    alt={movie.key}
                    className='movie__image'
                    data-testid='movieImage'
                />
                <p className='movie__title' data-testid='movieTitle'>{movie.name}</p>
            </div>
    </div>
    );
}

export default MovieThumbnail