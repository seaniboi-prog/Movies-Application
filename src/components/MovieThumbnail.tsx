import React from 'react'
import { useNavigate } from 'react-router-dom'
import { Movie } from '../models';
import './mainPage.scss';

interface Props {
    movie: Movie;
}

const MovieThumbnail:React.FC<Props> = ({movie}:Props) => {

    const navigate = useNavigate();

    const navigateToDetail = () => {
        navigate('/details', {state:{id: movie.id}})
    }

    return (
    <div className='movie__thumbnail'>
            <div className='movie__box'>
                <img
                    src={process.env.PUBLIC_URL + `movie_images/${movie.img}`}
                    alt={movie.key}
                    onClick={() => navigateToDetail()}
                    className='movie__image'
                />
                <p className='movie__title'>{movie.name}</p>
            </div>
    </div>
    );
}

export default MovieThumbnail