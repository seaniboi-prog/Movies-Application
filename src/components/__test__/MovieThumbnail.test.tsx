// import React from 'react'
// import ReactDOM from 'react-dom'
import MovieThumbnail from '../MovieThumbnail';
import { Movie } from '../../models';

import { render, cleanup, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import * as router from 'react-router'

describe('Movie Thumbnail Component Tests', () => {

    const dummyMovie:Movie = {
        id: 1,
        key: "key",
        name: "name",
        description: "description",
        genres: [],
        rate: "rate",
        length: "length",
        img: "img.jpg"
    }

    const mockNavigate = jest.fn()

    const renderComponent = () => {
        jest.spyOn(router, 'useNavigate').mockImplementation(() => mockNavigate);
        render(<MovieThumbnail movie={dummyMovie} />);
    }

    beforeEach(renderComponent);

    afterEach(cleanup);

    it("Renders Movie thumbnail correctly", () => {
        expect(screen.getByTestId('movieThumbnail')).toBeInTheDocument();
        expect(screen.getByTestId('movieBox')).toBeInTheDocument();
    });

    it("Renders Movie image correctly", () => {
        const imageElement:HTMLElement = screen.getByTestId('movieImage');
        expect(imageElement).toBeInTheDocument();
        expect(imageElement.getAttribute('alt')).toBe(dummyMovie.key);
        expect(imageElement.getAttribute('src')).toBe("movie_images/"+dummyMovie.img);
    });

    it("Renders Movie title correctly", () => {
        const titleElement:HTMLElement = screen.getByTestId('movieTitle');
        expect(titleElement).toBeInTheDocument();
        expect(titleElement).toHaveTextContent(dummyMovie.name);
    });

    // it("Redirects to Details Page after clicking", () => {
    //     expect(mockNavigate).toHaveBeenCalledWith('/details');
    // });

    it("Matches Snapshot", () => {
        const tree = renderer.create(<MovieThumbnail movie={dummyMovie}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
});