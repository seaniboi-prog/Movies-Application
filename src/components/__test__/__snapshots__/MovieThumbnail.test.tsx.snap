// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Thumbnail Component Tests Matches Snapshot 1`] = `
<div
  className="movie__thumbnail"
  data-testid="movieThumbnail"
>
  <div
    className="movie__box"
    data-testid="movieBox"
    onClick={[Function]}
  >
    <img
      alt="key"
      className="movie__image"
      data-testid="movieImage"
      src="movie_images/img.jpg"
    />
    <p
      className="movie__title"
      data-testid="movieTitle"
    >
      name
    </p>
  </div>
</div>
`;
